Add stack trace into make logs if DEBUG True (#994)
yogeshojha/rengine: README.md

![-----------------------------------------------------](https://raw.githubusercontent.com/andreasbm/readme/master/assets/lines/aqua.png)

### Submitting issues

> You can submit issue to the project but you should do it in a way that helps developers to resolve it as quickly as possible.

Reviewer: ```suggestion
You can submit issues related to this project, but you should do it in a way that helps developers to resolve it as quickly as possible.
```

> For that, you need to add as much valuable informations as possible.

Reviewer: ```suggestion
For that, you need to add as much valuable information as possible.
```

> You can have this valuable informations by doing this steps :

Reviewer: ```suggestion
You can have this valuable information by following these steps:
```

- Go to the root of the git cloned project

> - Edit `web/entrypoint.sh` and add at the top `export DEBUG=1`

Reviewer: ```suggestion
- Edit `web/entrypoint.sh` and add `export DEBUG=1` at the top
```

> This should give you this result
>   ```python

Reviewer: ```suggestion
This should give you this result

  ```python
```

  #!/bin/bash

  export DEBUG=1

  python3 manage.py migrate
  python3 manage.py runserver 0.0.0.0:8000

  exec "$@"
  ```

> - Restart the web container `docker-compose restart web`

Reviewer: ```suggestion
- Restart the web container: `docker-compose restart web`
```

> - To deactivate set **DEBUG** to **0** and restart web container again

Reviewer: ```suggestion
- To deactivate, set **DEBUG** to **0** and restart the web container again
```

Then, with **DEBUG** set to **1**, in the `make logs` output you could see the full stack trace to debug reNgine.

> Example with the tool arsenal version check API bug.
> ```
> web_1          |   File "/usr/local/lib/python3.10/dist-packages/celery/app/task.py", line 411, in __call__

Reviewer: ```suggestion
Example with the tool arsenal version check API bug.

```
web_1          |   File "/usr/local/lib/python3.10/dist-packages/celery/app/task.py", line 411, in __call__
```

web_1          |     return self.run(*args, **kwargs)
web_1          | TypeError: run_command() got an unexpected keyword argument 'echo'
```
Now you know the real error is `TypeError: run_command() got an unexpected keyword argument 'echo'`

And you can post the full stack trace to your newly created issue to help developers to track the root cause of the bug and correct the bug easily

> __Activating debug like this also give you the full stack trace in the browser__ instead of a 500 Error without any details.

Reviewer: ```suggestion
**Activating debug like this also give you the full stack trace in the browser** instead of an error 500 without any details.
```
For consistency reasons.

> So don't forget to open the developer console and check for any XHR request with 500 error.

Reviewer: ```suggestion
So don't forget to open the developer console and check for any XHR request with error 500.
```

> If there's any check the response of this request to get your detailed error.

Reviewer: ```suggestion
If there's any, check the response of this request to get your detailed error.
```
